tests/nodejs_install.etest

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env bash
+#
+# This program is free software: you can redistribute it and/or modify it under the terms of the Apache License
+# as published by the Apache Software Foundation, either version 2 of the License, or (at your option) any later
+# version.
+
+: ${EBASH_HOME:=$(dirname $0)/..}
+: ${EBASH:=${EBASH_HOME}/share}
+source "${EBASH}/ebash.sh" || { echo "Unable to source ${EBASH}/ebash.sh" ; exit 1 ; }
+
+# Import
+source "${EBASH_HOME}/install/nodejs"
+
+ETEST_clean_installation_check()
+{
+    emock "which" '
+    {
+        return 1
+    }'
+    check_for_node_installation
+}
+
+ETEST_cleanup_with_existing_symlinks()
+{
+    # Establish a test directory to install into
+    local test_prefix="$(mktemp -u usr_local.XXXXXX)"
```

Comment:
this is probably a lint error. I hate bash behavior on this but `local var=$(...)` __ALWAYS__ succeeds even if the subshell causes an error. Worse than that, is that it supresses our implicit fatal error handling. So you should not combine a local var assignment with a subhsell. So instead

```
local test_prefix
test_prefix="$(...)"
```

Reply:
I had no idea, thank you!